system-int-test/test/cmr/system_int_test/search/variable/variable_association_test.clj

```diff
@@ -0,0 +1,428 @@
+         [(format (str "Unable to create variable association on a collection revision and the whole "
+                       "collection at the same time for the following collections: %s.")
+                  c1-p1)]
+         response)))
+
+    (testing "Associate to non-existent collections"
+      (let [response (vu/associate-by-concept-ids
+                      token variable-name [{:concept-id "C100-P5"}])]
+        (vu/assert-variable-association-response-ok?
+         {["C100-P5"] {:errors ["Collection [C100-P5] does not exist or is not visible."]}}
+         response)))
+
+    (testing "Associate to deleted collections"
+      (let [c1-p1-concept (mdb/get-concept c1-p1)
+            _ (ingest/delete-concept c1-p1-concept)
+            _ (index/wait-until-indexed)
+            response (vu/associate-by-concept-ids
+                      token variable-name [{:concept-id c1-p1}])]
+        (vu/assert-variable-association-response-ok?
+         {[c1-p1] {:errors [(format "Collection [%s] does not exist or is not visible." c1-p1)]}}
+         response)))
+
+    (testing "ACLs are applied to collections found"
+      ;; None of PROV3's collections are visible
+      (let [response (vu/associate-by-concept-ids token variable-name [{:concept-id c4-p3}])]
+        (vu/assert-variable-association-response-ok?
+         {[c4-p3] {:errors [(format "Collection [%s] does not exist or is not visible." c4-p3)]}}
+         response)))
+
+    (testing "Variable association mixed response"
+      (let [response (vu/associate-by-concept-ids
+                      token variable-name [{:concept-id c2-p1}
+                                           {:concept-id "C100-P5"}])]
+        (vu/assert-variable-association-response-ok?
+         {["C1200000014-PROV1"] {:concept-id "VA1200000028-CMR"
+                                 :revision-id 1}
+          ["C100-P5"] {:errors ["Collection [C100-P5] does not exist or is not visible."]}}
+         response)))))
+
+(deftest associate-variable-failure-test
+  (e/grant-registered-users (s/context) (e/coll-catalog-item-id "PROV1"))
+  (let [variable-name "variable1"
+        variable (vu/make-variable {:Name variable-name})
+        token (e/login (s/context) "user1")
+        {:keys [concept-id revision-id]} (vu/create-variable token variable)
+        ;; The stored updated variable would have user1 in the originator id
+        variable (assoc variable :originator-id "user1")
+        coll-concept-id (:concept-id (d/ingest "PROV1" (dc/collection)))]
+    (testing "Associate variable using query sent with invalid content type"
+      (are [associate-variable-fn request-json]
+        (= {:status 400,
+            :errors
+            ["The mime types specified in the content-type header [application/xml] are not supported."]}
+           (associate-variable-fn token variable-name request-json {:http-options {:content-type :xml}}))
+        ; vu/associate-by-query {:provider "foo"}
```

Comment:
I am not sure if we have requirement to associate collections with variables by query. I suspect that we will, but not sure. I didn't want to make that decision now, so just commented out the code.